pkg/virt-api/webhooks/defaults.go

```diff
@@ -130,31 +130,6 @@ func setDefaultResourceRequests(clusterConfig *virtconfig.ClusterConfig, spec *v
 		resources.Requests[k8sv1.ResourceMemory] = resources.Limits[k8sv1.ResourceMemory]
 	}
 
-	if _, exists := resources.Requests[k8sv1.ResourceMemory]; !exists {
-		var memory *resource.Quantity
-		if spec.Domain.Memory != nil && spec.Domain.Memory.Guest != nil {
-			memory = spec.Domain.Memory.Guest
-		}
-		if memory == nil && spec.Domain.Memory != nil && spec.Domain.Memory.Hugepages != nil {
-			if hugepagesSize, err := resource.ParseQuantity(spec.Domain.Memory.Hugepages.PageSize); err == nil {
```

Comment:
If `spec.Domain.Memory.Guest` is nil but `spec.Domain.Memory.Hugepages` is not and the resource request does not exist, then the resource request was derived from the hugepages. So it treats the `spec.Domain.Memory.Hugepages.PageSize` as guest memory?